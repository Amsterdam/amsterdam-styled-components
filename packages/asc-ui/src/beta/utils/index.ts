import deprecatedWarning from './deprecatedWarning'

export {
  themeColor,
  srOnlyStyle,
  breakpoint,
  svgFill,
  fluidTypoStyle,
  typographyStyle,
  showAboveBackDrop,
  getValueFromTheme,
  themeSpacing,
} from './themeUtils'
export { valueFromObject } from './valueFromObject'
export { fromProps } from './fromProps'

export { deprecatedWarning }
