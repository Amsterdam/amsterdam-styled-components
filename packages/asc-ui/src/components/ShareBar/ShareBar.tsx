import React from 'react'
import ButtonBar, { Props as ShareBarProps } from '../ButtonBar'

type Props = ShareBarProps

const ShareBar: React.FC<Props> = ({ children, ...otherProps }) => (
  <ButtonBar {...otherProps}>{children}</ButtonBar>
)

export default ShareBar
