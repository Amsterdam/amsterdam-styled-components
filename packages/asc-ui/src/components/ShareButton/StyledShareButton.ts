import { AscCore } from '@datapunt/asc-core'
import styled from '../../styled-components'
import { getThemeColor } from '../../core/AscCoreUtils'

const StyledShareButton = styled(AscCore.Button.IconButton)<
  AscCore.Button.Props
>`
  padding: 0px;
  position: relative;
  justify-content: center;
  background: ${({ theme }) => getThemeColor(theme, 'tint', 'level5')}};

  &:focus,
  &:hover {
    background: ${({ hoverColor, theme }) =>
      hoverColor || getThemeColor(theme, 'secondary')};
  }
`

export default StyledShareButton
