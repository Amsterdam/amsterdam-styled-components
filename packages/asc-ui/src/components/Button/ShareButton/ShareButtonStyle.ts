import styled from '@datapunt/asc-core'
import IconStyle from '../../Icon/IconStyle'
import { color, svgFill } from '../../../utils'
import ButtonStyle, { Props as ButtonProps } from '../ButtonStyle'

export type Props = {
  hoverColor?: string
} & ButtonProps

const ShareButtonStyle = styled(ButtonStyle)<Props>`
  padding: 0;
  position: relative;
  justify-content: center;
  background: ${color('tint', 'level5')};

  &:focus,
  &:hover {
    background: ${({ hoverColor }) => hoverColor || color('secondary')};
  }

  ${IconStyle} {
    ${svgFill('tint', 'level1')};
  }
`

export default ShareButtonStyle
