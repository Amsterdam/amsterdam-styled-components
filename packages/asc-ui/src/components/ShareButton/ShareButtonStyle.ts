import styled from '@datapunt/asc-core'
import IconStyle from '../Icon/IconStyle'
import IconButtonStyle, {
  Props as IconButtonStyleProps,
} from '../IconButton/IconButtonStyle'
import { color, svgFill } from '../../utils'

export type Props = IconButtonStyleProps

const ShareButtonStyle = styled(IconButtonStyle)<Props>`
  padding: 0;
  position: relative;
  justify-content: center;
  background: ${color('tint', 'level5')};

  &:focus,
  &:hover {
    background: ${({ hoverColor }) => hoverColor || color('secondary')};
  }

  ${IconStyle} {
    ${({ theme }) => svgFill('tint', 'level1')({ theme })};
  }
`

export default ShareButtonStyle
