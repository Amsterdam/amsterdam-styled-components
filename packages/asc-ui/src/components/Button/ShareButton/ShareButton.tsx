import React from 'react'
import Icon from '../../Icon'
import ShareButtonStyle, {
  Props as ShareButtonStyleProps,
} from './ShareButtonStyle'

type Props = {
  iconSize?: number
} & ShareButtonStyleProps &
  React.HTMLAttributes<HTMLButtonElement>

const ShareButton: React.FC<Props> = ({
  children,
  onClick,
  iconSize,
  ...otherProps
}) => (
  <ShareButtonStyle
    data-testid="share-button"
    size={30}
    onClick={onClick}
    variant="blank"
    {...otherProps}
  >
    <Icon size={iconSize}>{children}</Icon>
  </ShareButtonStyle>
)

export default ShareButton
