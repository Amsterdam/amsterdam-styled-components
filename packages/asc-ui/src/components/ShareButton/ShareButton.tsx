import React from 'react'
import { Icon } from '../..'
import ShareButtonStyle, {
  Props as ShareButtonStyleProps,
} from './ShareButtonStyle'

type Props = {
  onClick?: React.MouseEventHandler<HTMLButtonElement>
  iconSize?: number
} & ShareButtonStyleProps

const ShareButton: React.FC<Props> = ({
  children,
  onClick,
  iconSize,
  ...otherProps
}) => (
  <ShareButtonStyle onClick={onClick} square {...otherProps}>
    <Icon size={iconSize}>{children}</Icon>
  </ShareButtonStyle>
)

export default ShareButton
