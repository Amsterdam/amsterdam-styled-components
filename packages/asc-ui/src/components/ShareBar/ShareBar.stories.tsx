import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { ReactComponent as Facebook } from '@datapunt/asc-assets/lib/Icons/Facebook.svg'
import { ReactComponent as Twitter } from '@datapunt/asc-assets/lib/Icons/Twitter.svg'
import { ReactComponent as Linkedin } from '@datapunt/asc-assets/lib/Icons/Linkedin.svg'
import { ReactComponent as Email } from '@datapunt/asc-assets/lib/Icons/Email.svg'
import { ReactComponent as Print } from '@datapunt/asc-assets/lib/Icons/Print.svg'
import ShareBar from './ShareBar'
import ShareButton from '../ShareButton'

const SimpleComponent: React.FC<{}> = ({ children }) => (
  <ShareBar>
    <ShareButton
      onClick={() =>
        window.open(
          `https://www.facebook.com/sharer/sharer.php?u=` +
            `${window.location.href}&title=${window.document.title}`,
          '_blank',
        )
      }
      hoverColor="#3b5999"
      iconSize={24}
      alignIcon="bottom"
    >
      <Facebook />
    </ShareButton>
    <ShareButton
      onClick={() =>
        window.open(
          `https://twitter.com/intent/tweet?url=${window.location.href}` +
            `&text=${window.document.title}`,
          '_blank',
        )
      }
      hoverColor="#55acee"
    >
      <Twitter />
    </ShareButton>
    <ShareButton
      onClick={() =>
        window.open(
          `
          https://www.linkedin.com/shareArticle?url=${window.location.href}` +
            `&mini=true&title=${window.document.title}`,
          '_blank',
        )
      }
      hoverColor="#0077B5"
    >
      <Linkedin />
    </ShareButton>
    <ShareButton
      onClick={() =>
        window.open(
          `mailto:?subject=${window.document.title.slice(0, 77)}` +
            `&title=${escape(window.location.href)}`,
        )
      }
    >
      <Email />
    </ShareButton>
    <ShareButton onClick={() => window.print()}>
      <Print />
    </ShareButton>
    {children}
  </ShareBar>
)

storiesOf('Composed/ShareBar', module).add(
  'default state with most used buttons',
  () => <SimpleComponent />,
)
