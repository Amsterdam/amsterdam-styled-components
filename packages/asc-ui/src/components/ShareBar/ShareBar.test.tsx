import * as React from 'react'
import { renderWithTheme } from '../../utils/withTheme'
import ShareBar from './ShareBar'
import 'jest-styled-components'
import ShareButton from '../ShareButton'

describe('ShareBar', () => {
  it('should render the share bar', () => {
    const component = renderWithTheme(
      <ShareBar>
        <ShareButton>One</ShareButton>
        <ShareButton>Two</ShareButton>
      </ShareBar>,
    )
    expect(component).toMatchSnapshot()
  })
})
