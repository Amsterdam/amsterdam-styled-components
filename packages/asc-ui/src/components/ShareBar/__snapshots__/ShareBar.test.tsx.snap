// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareBar should render the share bar 1`] = `
.c2 {
  display: block;
  width: 20px;
  height: 20px;
  padding: undefinedpx;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c2 > svg {
  width: inherit;
  height: inherit;
}

.c2 > svg rect,
.c2 > svg polygon,
.c2 > svg path {
  fill: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px;
}

.c0 > * {
  margin-right: 5px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  background: #ffffff;
  padding: 5px;
  width: 30px;
  height: 30px;
  padding: 0px;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #767676;
}

.c1:focus {
  outline-color: #FEC813;
  outline-style: solid;
  outline-width: medium;
}

.c1:hover {
  background: #E6E6E6;
}

.c1:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c1:focus,
.c1:hover {
  background: #ec0000;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <span
      class="c2"
      color="secondary"
    >
      One
    </span>
  </button>
  <button
    class="c1"
  >
    <span
      class="c2"
      color="secondary"
    >
      Two
    </span>
  </button>
</div>
`;
